pkg/blockstorage/vmware/vmware.go

```diff
@@ -251,38 +253,54 @@ var reVslmSyncFaultFatal = regexp.MustCompile("Change tracking invalid or disk i
 // SnapshotCreate is part of blockstorage.Provider
 func (p *FcdProvider) SnapshotCreate(ctx context.Context, volume blockstorage.Volume, tags map[string]string) (*blockstorage.Snapshot, error) {
 	var res types.AnyType
+	description := generateDescription(tags)
 	err := wait.PollImmediate(time.Second, defaultRetryLimit, func() (bool, error) {
-		log.Debug().Print("CreateSnapshot", field.M{"VolumeID": volume.ID})
-		task, lerr := p.Gom.CreateSnapshot(ctx, vimID(volume.ID), noDescription)
-		if lerr != nil {
-			return false, errors.Wrap(lerr, "CreateSnapshot task creation failure")
+		createErr := func() error {
```

Comment:
This is admittedly subjective, but I don't like the style of declaring functions within functions where not strictly necessary. Here the outer wait loop will be evaluating the function creation code each pass (though the compiler may optimize things).
I think it may be clearer if you were to define a set of helper functions before the wait loop and invoke them with arguments (instead of relying on lexical scoping)?  Perhaps even make these helpers methods on `FcdProvider` - could aid in unit testing!

Reply:
I don't know, honestly, on the other hand having function inside the loop helps understand what is happening in the code, you should jump to the function. Let's extract it to the helper and ask @ihcsim what does he think

Reply:
I am afraid I don't have much more thoughts re: unit tests. We should avoid conflating the scope of this PR. As previously [mentioned](https://github.com/kanisterio/kanister/pull/1555#discussion_r926078367), I proposed this refactoring because I felt it would make the code easier to read, and possibly more testable; didn't mean to imply that we add more tests in this PR. What you got still LGTM :+1:.

Also, I did it that way to make the diff easier to see. Other times, I used embedded functions or closures when I don't want to share the function with other callers, or avoid introducing a function with long list arguments. It's just another tool in the toolkit.